hack/git-validation.sh

```diff
@@ -5,5 +5,9 @@ if [ ! -x "$GIT_VALIDATION" ]; then
 	echo Try installing it with \"make install.tools\"
 	exit 1
 fi
-EPOCH_TEST_COMMIT=$(git merge-base ${DEST_BRANCH:-main} HEAD)
+
+EPOCH_TEST_COMMIT=$CIRRUS_BASE_SHA
+if [ -z "${CIRRUS_CI}" ]; then
```

Comment:
`CIRRUS_BASE_SHA` is, as we determined earlier this year, [worthless](https://github.com/containers/podman/pull/23081). 

I need to understand a lot more of this, like, what happened, what changed, why is `main` missing, what's the deal with `DEST_BRANCH`, and it's gonna take me a long time because I'm not familiar with this repo. Blocking until I get some time to understand.

Reply:
What a rabbit hole! TL;DR this appears to have been broken from the very beginning, and was only caught [by happenstance](https://github.com/containers/storage/pull/2064#pullrequestreview-2245280524) because the error was not (and still is not) causing a CI failure. 

Furthermore, the reason this works in podman is because of superconvoluted `clone_script` and other setup that has presumably been fine-tuned and tested over the years. We no longer have local expertise in such arcana. 

Proper solution is too complicated to get into here. It'll involve doing a git clone with proper branch tags for the given branch, and testing it under different branches, and under different conditions (CIRRUS_PR vs TAG vs MERGE vs PURPLEBUTTERFLIES or whatever) and different merge bases. I don't see a good ROI.

Good-enough solution: this approach, but please change as follows:
```diff
- if -z CIRRUS_CI
+ if -z EPOCH_TEST_COMMIT
```
Reason: yes, there's an implicit connection between `CIRRUS_BASE_SHA` and `CIRRUS_CI`, but it's much cleaner safer better to just actually test the actual actual, not some sort of distant cousin.